realm/realm-library/src/androidTest/java/io/realm/RealmTests.java

```diff
@@ -1056,6 +1057,44 @@ public void compactRealm_onExternalStorage() {
         Realm.deleteRealm(config);
     }
 
+    private Pair<Long, Long> populateTestRealmAndCompactOnLaunch(CompactOnLaunchCallback compactOnLaunch) {
+        final String REALM_NAME = "test.realm";
+        RealmConfiguration realmConfig = configFactory.createConfiguration(REALM_NAME);
+        Realm realm = Realm.getInstance(realmConfig);
+        populateTestRealm(realm, 100);
+        realm.close();
+        long before = new File(realmConfig.getPath()).length();
+        realmConfig = configFactory.createConfiguration(REALM_NAME, compactOnLaunch);
+        realm = Realm.getInstance(realmConfig);
+        realm.close();
+        long after = new File(realmConfig.getPath()).length();
+        return new Pair(before, after);
+    }
+
+    @Test
+    public void compactOnLaunch_shouldCompact() throws IOException {
```

Comment:
Added a test as the following:
https://github.com/realm/realm-java/pull/4857/commits/5244336c2241098b2c1b8dc205d436bfbe377e37#diff-eb98b3546f9b692ff01aaff903a3bdb3R1118